### PR TITLE
Parse match data into puuid to champion mapping

### PR DESCRIPTION
**Pull request by barun511**

### CODE REVIEW

File: data_service/src/match_processing/match_parser.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+use riven::models::match_v5::Match;
+
+use crate::types::PuuidToChampionMapping;
+
+pub async fn parse_match(match_data: Match) -> Vec<PuuidToChampionMapping> {
+    // Get match data.
```
Comment (barun511) [inline data_service/src/match_processing/match_parser.rs]: TODO(pyctal): Implement this function to get the test below to pass

Comment (pyctal) [reply data_service/src/match_processing/match_parser.rs]: On it

**Review by pyctal — APPROVED** [review]

**Comment by saikhurana98** [comment]
do we know if Match.metadata.participants and Match.info.participants are one-to-one mapped?
does the API guarantee this? it makes sense for it to but just wanted to confirm this

**Comment by saikhurana98** [comment]
there may be a massive oversight since I was told figuring out the mapping is not as straightforward.

**Comment by barun511** [comment]
We are guaranteed that the api 1:1 matches these.

Added two more tests for clarity, all lgtm

File: data_service/src/match_processing/match_parser.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        .info
+        .participants
+        .iter()
+        .map(|participant| PuuidToChampionMapping {
+            puuid: participant.puuid.clone(),
+            champion_name: participant.champion_name.clone(),
+        })
+        .collect();
+
+    // println!("Printing mappings");
```
Comment (barun511) [inline data_service/src/match_processing/match_parser.rs]: @saikhurana98 probably you should remove these comments

File: data_service/src/match_processing/match_parser.rs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    // for mapping in &to_return {
+    //     println!("puuid: {}, champion_name: {}", mapping.puuid, mapping.champion_name);
+    // }
+
+    to_return
```
Comment (barun511) [inline data_service/src/match_processing/match_parser.rs]: what's the point of this? Why not just return earlier?

File: data_service/src/match_processing/match_parser.rs [hunk 4]
```diff
@@ -0,0 +1,98 @@
+use riven::models::match_v5::Match;
+
+use crate::types::PuuidToChampionMapping;
+
+pub async fn parse_match(match_data: Match) -> Vec<PuuidToChampionMapping> {
+    // Get match data.
+
+    let to_return: Vec<PuuidToChampionMapping> = match_data
```
Comment (barun511) [inline data_service/src/match_processing/match_parser.rs]: nit: could leave out both the semicolon and the assignment ehre and just return

File: data_service/src/match_processing/match_parser.rs [hunk 5]
```diff
@@ -0,0 +1,98 @@
+use riven::models::match_v5::Match;
+
+use crate::types::PuuidToChampionMapping;
+
+pub async fn parse_match(match_data: Match) -> Vec<PuuidToChampionMapping> {
+    // Get match data.
```
Comment (barun511) [inline data_service/src/match_processing/match_parser.rs]: nit: get rid of this comment now that the implementation is done

File: compose.yml [hunk 6]
```diff
@@ -8,7 +8,7 @@ services:
       diesel setup &&
       diesel migration generate --diff-schema create_match_data &&
       diesel migration run;
-      cargo watch -x run"
+      cargo watch -x test"
```
Comment (barun511) [inline compose.yml:11]: Let's stick to test here for now, but will need to start thinking about how to set up different images for dev and prod

Comment (saikhurana98) [reply compose.yml:11]: noted, will read more about this

File: data_service/src/match_processing/match_parser.rs (hunk 2 above)
Comment (saikhurana98) [reply data_service/src/match_processing/match_parser.rs]: resolved, please check

File: data_service/src/match_processing/match_parser.rs (hunk 3 above)
Comment (saikhurana98) [reply data_service/src/match_processing/match_parser.rs]: resolved, please check

File: data_service/src/match_processing/match_parser.rs (hunk 4 above)
Comment (saikhurana98) [reply data_service/src/match_processing/match_parser.rs]: resolved, please check

File: data_service/src/match_processing/match_parser.rs (hunk 5 above)
Comment (saikhurana98) [reply data_service/src/match_processing/match_parser.rs]: resolved, please check

**Pull request merged**